src/strategy-heat-capacitor.js

```diff
@@ -0,0 +1,50 @@
+const { DateTime } = require("luxon");
+//const { loadDayData } = require("./utils");
+const { run_buy_sell_algorithm, find_temp } = require("./strategy-heat-capacitor-functions");
+//const { difference } = require("lodash");
+
+module.exports = function (RED) {
+  function TempMan(config) {
+    RED.nodes.createNode(this, config);
+    const node = this;
+
+    node.time_heat_1c = Number(config.time_heat_1c);
+    node.time_cool_1c = Number(config.time_cool_1c);
+    node.max_temp_adjustment = Number(config.max_temp_adjustment);
+    node.min_saving_NOK_kWh = Number(config.min_saving_NOK_kWh);
+    // sanitise disabled output as this is used when all else fails
+    if (isNaN(node.disabled_op)) {
+      node.disabled_op = 0;
+    }
+
+    this.on("input", function (msg) {
+      // Using msg.* to change specific TM property.
+      if (msg.hasOwnProperty("time_heat_1c")) node.time_heat_1c = Number(msg.time_heat_1c);
+      if (msg.hasOwnProperty("time_cool_1c")) node.time_cool_1c = Number(msg.time_cool_1c);
+      if (msg.hasOwnProperty("max_temp_adjustment")) node.max_temp_adjustment = Number(msg.max_temp_adjustment);
+      if (msg.hasOwnProperty("min_saving_NOK_kWh")) node.min_saving_NOK_kWh = Number(msg.min_saving_NOK_kWh);
+
+      if (msg.topic == "configure") {
+        node.setpoint = Number(msg.payload);
+      } else {
```

Comment:
If I understand this right, it is about sending config as input.
To avoid confusing the users, you should do this the same way as it is done in the other nodes. That is sending the config in a config-object in the payload.

I believe the setpoint also could be sent as config. How about having the setpoint in the config dialog, to set a default value? Or maybe that does not make sense? (I do not fully understand this node, yet :-)

Config-variables sent in the input should definitely have camelCasing as in other nodes, since they are visible to the user.

You should be able to use the `getEffectiveConfig` function in `utils.js` to read the config. It will give you the config either from the message or from the node. It requires that you save the config to the nodes context when the node is loading. See `strategy-lowest-price.js` lines 9-20 how this is done, and how it is used in `handle-input.js` line 6.

Some people are sending config and priceData in the same message, so that should work here too.

Reply:
> I believe the setpoint also could be sent as config. How about having the setpoint in the config dialog, to set a default value? Or maybe that does not make sense? (I do not fully understand this node, yet :-)

Good point. It would simplify the flow... I've tried to think of a situation where one would like to keep the current design, however, didn't find any very good arguments. Now... Where to put it.. I am tempted to use msg.topic = "config", however, in our current setup it should be msg.config.setpoint... Seems simpler to use .topic, but we should probably follow convention?

![image](https://user-images.githubusercontent.com/49571281/152611752-8f9416df-5d87-4908-a932-5c797f0ebb6d.png)

Reply:
Yes, please follow the save convention as already used. Important that users see consistency. Topic is ment for message topics, like used in MQTT, and is also suited for entity_id as in home assistant. Data could have been used, but I think it is easier when everything you need is in the payload.

But convention is most important anyway.

Reply:
Done